crates/sui-core/src/authority_server.rs

```diff
@@ -393,6 +394,26 @@ impl ValidatorService {
             SuiError::InvalidSystemTransaction.into()
         );
 
+        // Enforce overall transaction size limit.
+        let tx_size = bcs::serialized_size(&certificate.data()).map_err(|e| {
+            SuiError::TransactionSerializationError {
+                error: e.to_string(),
+            }
+        })?;
+        let max_tx_size_bytes = epoch_store.protocol_config().max_tx_size_bytes();
+        fp_ensure!(
+            tx_size as u64 <= max_tx_size_bytes,
+            SuiError::UserInputError {
+                error: UserInputError::SizeLimitExceeded {
+                    limit: format!(
+                        "serialized transaction size exceeded maximum of {max_tx_size_bytes}"
+                    ),
+                    value: tx_size.to_string(),
+                }
+            }
+            .into()
+        );
```

Comment:
this code is duplicated with the code in handle_transaction - can you factor it out?

Reply:
I've moved it into a validity check implemented for `SenderSignedData` which also calls verify user input and `TransactionData`'s validity checks. cc @mwtian